packages/editor/src/extensions/link/helpers/pasteHandler.ts

```diff
@@ -20,32 +20,65 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 import { Editor } from "@tiptap/core";
 import { MarkType } from "@tiptap/pm/model";
 import { Plugin, PluginKey } from "@tiptap/pm/state";
+import linkifyHtml from "linkify-html";
 import { find } from "linkifyjs";
+import { linkRegex } from "../link";
 
 type PasteHandlerOptions = {
   editor: Editor;
   type: MarkType;
+  linkOnPaste: boolean;
 };
 
 export function pasteHandler(options: PasteHandlerOptions): Plugin {
+  let shiftKey = false;
   return new Plugin({
     key: new PluginKey("handlePasteLink"),
     props: {
+      handleKeyDown(_, event) {
+        shiftKey = event.shiftKey;
+        return false;
+      },
       handlePaste: (view, event, slice) => {
+        const clipboardHtmlData = event.clipboardData?.getData("text/html");
+        if (clipboardHtmlData) {
+          return false;
+        }
+
         const { state } = view;
         const { selection } = state;
         const { empty } = selection;
 
-        if (empty) {
-          return false;
-        }
-
         let textContent = "";
 
         slice.content.forEach((node) => {
           textContent += node.textContent;
         });
 
+        // don't deal with markdown links in this handler
+        if (linkRegex.test(textContent)) {
+          // reset the regex since we don't want the above test method to affect other places where the regex is used
+          linkRegex.lastIndex = 0;
+          return false;
+        }
+
+        if (shiftKey) {
+          return false;
+        }
```

Comment:
Reset the `shiftKey` flag here otherwise links will be ignored every time.

Reply:
https://github.com/streetwriters/notesnook/blob/2878ef32b422ead2235340ff137d0cbc013e86e4/packages/editor/src/extensions/link/helpers/pasteHandler.ts#L39

`event.shiftKey` is either `true` or `false`, so we don't have to reset manually

Reply:
What if someone pastes using `Ctrl + Shift + V` and then pastes again using the context menu? In that case the `shiftKey` will still be true.

Reply:
Nice observation, thanks